### ISSUE TITLE
Hey your data loader is kind of slow. 

### ISSUE THREAD

**Issue by shamanez** [lukysummer/WaveGAN-in-PyTorch#1]
I think it would be nice if you can load only the paths of file names when loading the data set without loading the numpy arrays.  

class FlatDirectoryAudioDataset(tdata.Dataset): #cutomized dataloader

    def __init__(self, data_dir, transform=None):
        self.data_dir = data_dir
        self.transform = transform
        self.files = self.__setup_files()

    def __len__(self):
        """
        compute the length of the dataset
        :return: len => length of dataset
        """
        return len(self.files)

    def __setup_files(self):

        file_names = os.listdir(self.data_dir)
        files = []  # initialize to empty list

        for file_name in file_names:
           
            possible_file = os.path.join(self.data_dir, file_name)
            if os.path.isfile(possible_file) and (file_name.lower().endswith('.wav') or file_name.lower().endswith('.mp3')): #&& (possible_file.lower().endswith('.wav') or possible_file.lower().endswith('.mp3')):
                files.append(possible_file)

        # return the files list

        return files

    def __getitem__ (self,index):

        sample, _ = librosa.load(self.files[index], 16000)
        

        if self.transform:
            sample=self.transform(sample)
            

        sample = torch.from_numpy(sample)    
        return sample

class transform_audio(object):  #Cutomized tranformation function or preprocessing
    def __init__(self, total_length=64000, normalize=True):
     
        self.audio_length = total_length
        self.normalize = normalize

    def __call__(self,signal):

        
        ##### 1. Change the datatype as float #####
        if signal.dtype == np.int16:
            signal = signal.astype(np.float32)
            signal /= 32768.

    

        ##### 2. Pad the audio file to match its length with audio_length #####
        if len(signal) < self.audio_length: 
            delta = self.audio_length - len(signal)
            signal = np.pad(signal, (0, delta), mode = 'edge')

        else:
            signal = signal[:self.audio_length]

        ##### 3. Reshape the signal #####
        n_channels = 1 if signal.ndim == 1 else signal.shape[1]        
        signal = signal.reshape(n_channels, signal.shape[0])
            
        ##### 4. Normalize the signal #####
        if self.normalize:
            factor = np.max(np.abs(signal))
            if factor>0:
                signal /= factor

        return signal

    def __repr__(self):
        return self.__class__.__name__ + '(total_length={64000},signal,normalize={True})'.format(self.audio_length,self.normalize)
